File: Leetcode/Daily/21-05-2024/prob.go
func subsets(nums []int) [][]int {
    var ans [][]int

    sz := len(nums)

    for i:=0;i<(1<<sz);i++{
        var v []int
        for j:=0;j<sz;j++{
            if (i>>j)&1 == 1{
                v = append(v,nums[j])
            }
        }
        ans = append(ans,v)
    }

    return ans
}

File: Leetcode/Daily/21-05-2024/prob.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        int sz = pow(2,nums.size())-1;
        vector<vector<int>>ans;
        for(int i=0;i<=sz;i++){
            vector<int>res;
            for(int j=0;j<nums.size();j++){
                if((i>>j)&1){
                    res.push_back(nums[j]);
                }
            }

            ans.push_back(res);
        }


        return ans;
    }
};